src/Dofs/DofHandler.jl

```diff
@@ -849,8 +849,8 @@ function _evaluate_at_grid_nodes(dh::DofHandler, u::Vector{T}, fieldname::Symbol
         CT = getcelltype(get_grid(dh), first(sdh.cellset))
         ip_geo = default_interpolation(CT)
         local_node_coords = reference_coordinates(ip_geo)
-        qr = QuadratureRule{getrefshape(ip)}(zeros(length(local_node_coords)), local_node_coords)
         ip = getfieldinterpolation(sdh, field_idx)
+        qr = QuadratureRule{getrefshape(ip)}(zeros(length(local_node_coords)), local_node_coords)
```

Comment:
Note: This bug was detected while running the example, and I decided to fix it here instead of doing a separate PR since it is such a small change.

Reply:
Could you also add a regression test on this?

Reply:
Done

Reply:
Note to when this is merged; this bug was now fixed and tested by #800. 

Make sure to not duplicate fix and tests. (I can merge with master when someone tells me they are ready to review this one)